File: src/main/java/ru/lionzxy/ampersand/ExperimentMain.java
package ru.lionzxy.ampersand;

import com.fazecast.jSerialComm.SerialPort;

import java.nio.ByteBuffer;
import java.util.Random;

public class ExperimentMain {
    private static final Random RANDOM = new Random();

    public static void main(String... args) throws InterruptedException {
        ChartGUI chartGUI = new ChartGUI();
        chartGUI.start();

        double baseline = 0.442;
        while (true) {
            int iterationNear = RANDOM.nextInt(3000);
            baseline = baseline + RANDOM.nextDouble() * 0.1 - 0.05;
            for (int i = 1; i <= iterationNear; i++) {
                double newValue;

                if (RANDOM.nextDouble() < 0.1) {
                    newValue = (baseline + RANDOM.nextDouble() * 0.5 - 0.25) * 1000;
                } else {
                    newValue = (baseline + RANDOM.nextDouble() * 0.1 - 0.05) * 1000;
                }
                chartGUI.addValue((int) newValue);
                Thread.sleep(1L);
            }
        }
    }

}

File: src/main/java/ru/lionzxy/ampersand/ExperimentMain.kt
package ru.lionzxy.ampersand

import ru.lionzxy.ampersand.chart.ChartPoint
import java.util.*
import kotlin.random.Random

fun main(args: Array<String>) {
    val chartGUI = UpdateLooper()
    chartGUI.start()

    var baseline = 0.442
    while (true) {
        val iterationNear = Random.nextInt(3000)
        baseline = baseline + Random.nextDouble() * 0.1 - 0.05
        for (i in 1..iterationNear) {
            val newValue: Double

            if (Random.nextDouble() < 0.1) {
                newValue = (baseline + Random.nextDouble() * 0.5 - 0.25)
            } else {
                newValue = (baseline + Random.nextDouble() * 0.1 - 0.05)
            }
            chartGUI.addValue(
                ChartPoint(
                    voltage = newValue,
                    current = newValue
                )
            );
            Thread.sleep(1L)
        }
    }
}
